tests/unit/graph_reader/test_artifact_collection.py

```diff
@@ -174,3 +176,55 @@ def test_two_sessions(
                 assert art_pred_session_line_index >= 0
                 # required art session is in from of art session
                 assert art_pred_session_line_index < art_session_line_index
+
+
+@pytest.mark.parametrize(
+    "input_parameters, input_values, expected_values",
+    [
+        pytest.param([], dict(), {"add_p": "ap"}, id="no_input"),
+        pytest.param(["a", "p"], dict(), {"add_p": "ap"}, id="default_value"),
+        pytest.param(
+            ["a", "p"],
+            {"a": "A", "p": "P"},
+            {"add_p": "AP"},
+            id="overriding_value",
+        ),
+    ],
+)
+def test_module_run(
+    linea_db, execute, input_parameters, input_values, expected_values
+):
+    """
+    Test the module generated from ArtifactCollection can run with/without
+    input parameters
+    """
+
+    code = """\n
+import lineapy
+a = "a"
+p = "p"
+b = a+p
+lineapy.save(b,'add_p')
+"""
+    execute(code, snapshot=False)
+    artifact_list = ["add_p"]
+    ac = ArtifactCollection(
+        linea_db, artifact_list, input_parameters=input_parameters
+    )
+    temp_folder = tempfile.mkdtemp()
+    temp_module_path = pathlib.Path(
+        temp_folder, "artifactcollection_module.py"
+    )
+    with open(temp_module_path, "w") as f:
+        f.writelines(ac.generate_module_text())
+
+    cmds = ["python", str(temp_module_path)]
+    # Add input parameter values if specified in cmds
+    for par, val in input_values.items():
+        cmds += [f"--{par}", val]
+
+    p = subprocess.run(cmds, capture_output=True, text=True)
+    assert p.returncode == 0
+    module_run_result = eval(p.stdout)
```

Comment:
not entirely sure how you are doing this but I feel there is a better alternative. 1) don't like depending on eval if I can help it. 2) depends on the output of a command. we can directly get access to the value in the artifact store (with aubhro's changes, each test can run on its own version of the .lineapy folder)

Reply:
Here, we are validating that the functions defined in the module itself are correct by running a subprocess call. The only thing we have is the stdout/stderror. We either need to convert the stdout to a python object(this is where eval comes in)  and compare the object in python or we need to construct what the stdout looks as a string to compare (don't want to deal with the line split in stdout neither).

Reply:
let move ahead for now. resolving this one.